evol/population.py

```diff
@@ -151,16 +152,19 @@ def survive(self, fraction=None, n=None, luck=False) -> 'Population':
         :return: self
         """
         if fraction is None:
-            if n is None:
-                raise ValueError('everyone survives! must provide either "fraction" and/or "n".')
+            if (n is None) or (n > len(self.individuals)):
```

Comment:
In the future we'll have populations that can have islands which will allow for populations that are larger than the intended size. I do agree that if `n>intended_size` that we should send some form of warning.

Reply:
This was intentional: the size of a population should be rather well defined when going through an evolution, but the size of an incoming population is completely free. I do not want to raise any exceptions when someone passes a smaller population than expected. If however `n > self.intended_size` we may want to raise.